cmd/test/main.go

```diff
@@ -357,16 +372,30 @@ func TestWrapper(ctx *TestCommandContext, args []string, testCommandFlags *TestC
 
 func test(ctx *TestCommandContext, paths []string, testCommandData *TestCommandData) error {
 	if paths[0] == "-" {
-		tempFile, err := os.CreateTemp("", "datree_temp_*.yaml")
-		if err != nil {
-			return err
-		}
-		defer os.Remove(tempFile.Name())
+		if testCommandData.SaveRendered != "" {
+			content := []byte{}
+			_, err := os.Stdin.Read(content)
+			if err != nil {
+				return err
+			}
 
-		if _, err := io.Copy(tempFile, os.Stdin); err != nil {
-			return err
+			savedFilenam, err := ctx.CommandRunner.SaveRenderedFile(testCommandData.SaveRendered, content)
```

Comment:
I think this `SaveRenderedFile` function can receive `io.Reader` as an argument type instead of the content `[]byte`.
This way you don't have to save anything in memory. Think of huge helm charts that have to be saved in memory just to be written again in the provided path.